src/js/providers/html5.ts

```diff
@@ -179,11 +179,18 @@ function VideoProvider(this: HTML5Provider, _playerId: string, _playerConfig: Ge
         },
 
         canplay(this: ProviderWithMixins): void {
-            _canSeek = true;
-            if (!_androidHls) {
-                _setMediaType();
+            if (OS.iOS && Browser.safari) {
+                return;
             }
-            checkVisualQuality();
+            _canPlayHandler();
+            VideoEvents.canplay.call(_this);
```

Comment:
`VideoEvents.canplay` should be called on `canplay` for all browsers. It should no be called on `canplaythrough`. If we break down what it does there are two things it does:

```
    canplay(this: ProviderWithMixins): void {
        // If we're not rendering natively text tracks will be provided from another source - don't duplicate them here
        if (this.renderNatively) {
            this.setTextTracks(this.video.textTracks);
        }
        this.trigger(MEDIA_BUFFER_FULL);
    },
```

The things relevant to this issue are `_canSeek = true` and triggering `MEDIA_BUFFER_FULL`. Those can wait until `canplaythrough` on iOS. The captions and quality methods should not be deferred.

Reply:
Also what does "bufferfull" have to do with anything? Does anything listen for that event?